### PR TITLE
[DO NOT MERGE] pro: link against libsodium on Linux

### PR DESCRIPTION
**Pull request by radfish**
This is needed to link against libmonero_wallet from monero
with PR #4159 merged.

Please merge AFTER: https://github.com/monero-project/monero/pull/4159 is merged

Not touching other platforms because can't test.

### CODE REVIEW

**Comment by cryptochangements34** [comment]
maybe mark [DO NOT MERGE] in the title then change it back after #4159 is merged?

**Comment by stoffu** [comment]
I believe linking to libsodium is already necessary since the merge of #3999 which starts to use the library function `crypto_verify_32`.

File: monero-wallet-gui.pro [hunk 1]
```diff
@@ -116,6 +116,7 @@ LIBS += -L$$WALLET_ROOT/lib \
         -lepee \
         -lunbound \
         -leasylogging \
+        -lsodium \
```
Comment (MoroccanMalinois) [inline monero-wallet-gui.pro]: Since it's also required for android, could you please move that addition to line 292 ?

**Comment by BigslimVdub** [comment]
@stoffu have you checked this addition on OS X when building?

**Comment by stoffu** [comment]
@BigslimVdub 
Yes, this is needed for Aeon, too. https://github.com/aeonix/aeon-gui/pull/11

**Pull request closed without merging**